source-code/app/src/main/java/org/buildmlearn/toolkit/fragment/DraftsFragment.java

```diff
@@ -397,4 +408,22 @@ private void deleteAllDrafts() {
     private void restoreSelectedView() {
         restoreColorScheme();
     }
+
+    @Override
+    public void onBackPressed() {
+
+        if (mAdapter.selectedPositionsSize() != 0){
+            for(int i=0;i<mAdapter.getCount();i++)
+                if(mAdapter.isPositionSelected(i)) {
+                    mListView.getChildAt(i).setBackgroundColor(0);
+                    mAdapter.removeSelectedPosition(i);
+                }
+            restoreColorScheme();
+        }
+        else{
+            FragmentManager fragmentManager = getFragmentManager();
+            fragmentManager.beginTransaction().setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
```

Comment:
In this case, what I see that it can be easily done by adding a check in ```onBackPressed```, if some selected item is there, then we will deselect it, else perform the usual ```onBackPressed``` by calling ```super.onBackPressed()```. If you want you can fix with this hints.